### ISSUE TITLE
Unable to use Group Managed Service Account as Service Login

### ISSUE THREAD

**Issue by mmcapps88** [cjmurph/PmsService#59]
Hello. I run all my computers as part of a domain and users and computers are, in turn, members of that domain.

I wish to run the Plex Media as a Service (Service) under the login for a specially created Group Managed Service Account under Active Directory.

At the point of installing Plex Media as a Service, it is impossible to select Service Accounts as the login for the service as only the "Users" object can be selected so I then tried to change the service login through Windows Services.

Although I am able to select my Group Managed Service Account from the list of available Services, when I do select it and try to apply the changes, I get an error stating that the password is invalid. Since it is a Group Managed Service Account, I have no control over the password, which is generated at the time the Group Managed Service Account is created.

Would it be possible to enhance the application to make it capable of allowing the use use of Domain member Group Managed Service Accounts as the login for the Plex Service?

Thanks and regards,

Patrick

![ManagedServiceAccount-Selected-As-Logon-Account](https://user-images.githubusercontent.com/6311670/154997874-83c4933f-cb4c-4be0-b8da-33ef031a95f9.jpg)
![ManagedServiceAccount-Selected-As-Logon-Error](https://user-images.githubusercontent.com/6311670/154997876-aa31238c-bd64-46de-88be-8b10745d1d29.jpg)
.

**Comment by cjmurph** [comment]
I can have a look at modifying the installer to allow this. You should be able to do this manually through services.msc though. you shouldn't need a password, just choose the account. The account should then be automatically granted log on as a service privelge.

**Comment by cjmurph** [comment]
did you manage to get the service to run through the services snap in?

**Comment by mmcapps88** [comment]
Hello. No. I ended up setting up the service to run under my own user identity and password and I missed your update about modifying the installer. I have just changed the logon for the service through the services.msc to have it use the managed service account.

At first, it seemd as if it would not work again but then I noticed that Windows was populating the password fields or the password from my own account was still left in place so I deleted the password from the two password entry fields and applied the change. I was then presented with the following dialogue, which appears to show that the change was successful.

To confirm that the service will run with the newly applied managed service account, I restarted it and it is running. However, it has now lost all of the configuration I had previously set up so it no longer knows about my libraries, DVR settings, recording schedules etc. In addition, it cannot see any of my mapped drives, which contain all of my media and which are on an external NAS. I tried adding Drive Mappings via your applications settings but these are not seen by the Plex application when I try to add back my libraries.

I shall now have to try to figure out how to recover all of my settings and media meta-data.

![Set Plex Service Log On with Managed Service Account](https://user-images.githubusercontent.com/6311670/212136627-dcce54c5-3266-4e68-aec6-5c1d06a025ad.JPG)

**Comment by cjmurph** [comment]
Hi,

the reason for this is that plex builds its library per user. you can configure it to put the library in a common place though, which is my personal preference. You could move the library from your own user folder to somewhere common (e.g. C:\Plex) and then configure plex to use that location for whichever user plex starts as.

**Comment by mmcapps88** [comment]
Hi. Yes. I have now remembered reading about this issue with switching the user account, which is used by Plex. The problem I now have is the fact that I would need to be able to log on to the machine as the managed service account user in order to move the Plex files from my user account %APPDATA% folder to the service account's %APPDATA% folder as set out in this article https://www.plexopedia.com/plex-media-server/windows/moving-plex-media-server-different-user-account/

As yet, I have not found a way to log on as the managed service user nor have I been able to carry out the suggested registry edits for that user account.

I fear I may have to switch back to using my own login for the service but thanks for your help on this. I realise my current problems are not related to your application so you can close this issue if you like.

**Comment by cjmurph** [comment]
No, you can move the Plex data folder from your user account to a non user location, like C:\plex for example. Then start plex as the managed service, go to the settings in plex and change the data folder location to be wherever you moved it, C:\Plex in this example.

I'm not sure if you change that setting before or after you move the data, there is a plex doc about it somewhere.

**Comment by cjmurph** [comment]
@mmcapps88 Is this ok to close?

**Comment by mmcapps88** [comment]
Hi,

Thank you. Yes you can.

Regards,

Patrick
